packages/app/src/composables/useSpecFilter.ts

```diff
@@ -0,0 +1,43 @@
+import { useSpecStore } from '../store'
+import { useMutation, gql } from '@urql/vue'
+import { SpecsList_SetSpecFilterDocument } from '@packages/app/src/generated/graphql'
+import { ref, watch } from 'vue'
+import { useDebounce } from '@vueuse/core'
+
+gql`
+mutation SpecsList_SetSpecFilter($specFilter: String!) {
+  setSpecFilter(specFilter: $specFilter)
+}
+`
+
+export function useSpecFilter (savedFilter?: string) {
+  const specStore = useSpecStore()
+  const saveSpecFilter = useMutation(SpecsList_SetSpecFilterDocument)
+
+  // prefer a filter from client side store, saved filter in gql can be stale
+  // and is only used to set the value in the store on first load
+  const initialFilter = specStore.specFilter ? specStore.specFilter : savedFilter ?? ''
```

Comment:
These were not equivalent since the `specFilter` in the store was initialized as an empty string (so, not nullish). Buuut I think it is probably better to distinguish the initial state in the store from the state set by this function, so I've set the initial state to `undefined`.